model/src/main/kotlin/licenses/ResolvedLicenseInfo.kt

```diff
@@ -291,6 +294,24 @@ fun List<ResolvedLicense>.filterExcluded() =
                 resolvedLicense.locations.any { it.matchingPathExcludes.isEmpty() }
     }.map { it.filterExcludedCopyrights() }
 
+/**
+ * Filter licenses for which a resolution exists. Licenses are removed if all the conditions match:
+ *  * Any of the [resolutions] matches the [violations].
+ *  * The violation was caused by a [license][RuleViolation.license].
+ *  * The package with the [packageId] is licensed with the license from the violation.
+ */
+fun List<ResolvedLicense>.filterLicenseResolutions(
```

Comment:
Extend `Collection<ResolvedLicense>` rather than `List` to make this more generic.